internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go

```diff
@@ -260,11 +260,22 @@ func resourceCloudProviderAccessAuthorizationStateUpgradeV0(ctx context.Context,
 }
 
 func authorizeRole(ctx context.Context, client *admin.APIClient, d *schema.ResourceData, projectID string, targetRole *admin.CloudProviderAccessRole) diag.Diagnostics {
-	req := &admin.CloudProviderAccessRole{
+	var roleID string
+	providerName := targetRole.ProviderName
+	if providerName == constant.AZURE {
+		roleID = targetRole.GetId()
+	} else {
+		roleID = targetRole.GetRoleId()
+	}
+	req := &admin.CloudProviderAccessRoleRequestUpdate{
```

Comment:
Relevant MMS changes:
1. [PR of one of the changes to updates](https://github.com/10gen/mms/pull/106717/files)
2. [PR of one of the changes to list](https://github.com/10gen/mms/pull/104901/files)

Reply:
after rebasing from the merged master SDK this is no longer relevant